src/ucp/wireup/select.c

```diff
@@ -428,14 +421,25 @@ ucp_wireup_select_transport(const ucp_wireup_select_params_t *select_params,
                 continue;
             }
 
-            ucp_wireup_update_select_info(context, md_attr, rsc_index,
-                                          iface_attr, &ae->iface_attr,
-                                          addr_index, criteria, &sinfo);
+            score        = criteria->calc_score(context, md_attr, iface_attr,
+                                                &ae->iface_attr);
+            priority     = iface_attr->priority + ae->iface_attr.priority;
+
+            if (!is_reachable) {
+                is_reachable = 1;
+                ucp_wireup_init_select_info(context, score, addr_index,
+                                            rsc_index, priority,
+                                            criteria->title, &sinfo);
+            } else {
+                ucp_wireup_update_select_info(context, score, addr_index,
```

Comment:
i think "ucp_wireup_update_select_info" function is redundant
do it here like this:
if (!found || (res > 0) || (res == 0 && priority>select_info->priority) {
 ucp_wireup_init_select_info(..)
 found = 1;
 }

Reply:
then 
1) need to call `ucp_score_cmp(score, sinfo->score)` with uninitialized ` sinfo->score` or initialized with some meaningless value before `if (...)` statement to suppress static checkers.
2) call it inside like `if(!found || ((ucp_score_cmp(...), (res > 0 || res ==0) ...))`
3) have smth like 
```
if (!found ) {
    ucp_wireup_init_select_info();
    found = 1;
} else {
    res = ucp_score_cmp();
    if (res...) {
        ucp_wireup_init_select_info();
    }
}
```

IMO all these don't add readability

Reply:
something like "ucp_wireup_is_better"

Reply:
`int ucp_wireup_score_cmp(score1, prio1, score2, prio2)`?

Reply:
> `int ucp_wireup_score_cmp(score1, prio1, score2, prio2)`?

👍

Reply:
how would it help with uninitialized fields?

Reply:
if (!found || ucp_wireup_score_cmp(score1, prio1, score2, prio2)) {
  update-select-info
}

Reply:
Then we assume that priority can't be equal, right? Also I think some analyzers can generate false positive here.

Reply:
priority can be equal, but it was also the case before this PR..

Reply:
IMO it's too sensitive. minor changing behaviour of `ucp_wireup_score_cmp` for case `((score1 == score2) && (prio1 == prio2))` may encapsulate non-obvious selection change.